### PR TITLE
Do not fail if the runner does not have a strand in serializer

### PR DESCRIPTION
**Pull request by enescakir**
Usually, we don't expect that the runner model will lack a strand. However, due to eager loading sending multiple queries, there may be brief periods when the runner doesn't have a strand in the loaded list. This PR ensures that the serializer doesn't fail if the runner is without a strand.

With recent changes, this state isn't only the state of virtual machine. So I moved to the runner model as "display_state" similar to the other models.

I also rename variables to make the test more readable.

### CODE REVIEW

File: serializers/web/github_runner.rb [hunk 1]
```diff
@@ -20,9 +20,9 @@ def self.base(runner)
       } : nil,
       vm_state: if runner.vm
                   runner.vm.display_state
-                elsif runner.strand.label == "wait_vm_destroy"
+                elsif runner.strand&.label == "wait_vm_destroy"
```
Comment (fdr) [inline serializers/web/github_runner.rb:23]: It's a bit smelly to have multiple "safe navigation" like this. Should `runner` learn a `display_state`, or any other refactoring you can think of? Or is the cure worse than the disease?

A local variable might also be fine.

**Review by fdr — COMMENTED** [review]
.

File: serializers/web/github_runner.rb (hunk 1 above)
Comment (enescakir) [reply serializers/web/github_runner.rb]: Makes sense. I focused too much on these lines.

I refactored now. With recent changes, this state isn't only the state of virtual machine. So I moved to the runner model as "display_state" similar to the other models.

File: model/github_runner.rb [hunk 2]
```diff
@@ -25,6 +25,15 @@ def runner_url
     "http://github.com/#{repository_name}/settings/actions/runners/#{runner_id}" if runner_id
   end
 
+  def display_state
+    return vm.display_state if vm
+    if strand
+      return "deleted" if strand.label == "wait_vm_destroy"
+      return "reached_concurrency_limit" if strand.label == "wait_concurrency_limit"
+    end
+    "not_created"
```
Comment (fdr) [inline model/github_runner.rb:34]:
```suggestion 
    case strand&.label
    when "wait_vm_destroy"
      "deleted"
    when "wait_concurrency_limit"
      "reached_concurrency_limit"
    else
      "not_created"
    end
```

It blows up the line count, but probably has fewer expressions and definitely less flow control. Not entirely coincidentally, it's half the width.

**Review by fdr — CHANGES_REQUESTED** [review]
presuming I retained all the semantics correctly

File: model/github_runner.rb (hunk 2 above)
Comment (enescakir) [reply model/github_runner.rb]: Another alternative which doesn't blow up line counts

```
    case strand&.label
    when "wait_vm_destroy" then "deleted"
    when "wait_concurrency_limit" then "reached_concurrency_limit"
    else "not_created"
    end
```

Comment (fdr) [reply model/github_runner.rb]: Sure :)

**Review by bsatzger — APPROVED** [review]

File: views/project/github.erb [hunk 3]
```diff
@@ -32,7 +32,7 @@
               <th scope="col" class="py-3.5 pl-4 pr-3 text-left text-sm font-semibold text-gray-900 sm:pl-6">Runner</th>
               <th scope="col" class="px-3 py-3.5 text-left text-sm font-semibold text-gray-900">Repository</th>
               <th scope="col" class="px-3 py-3.5 text-left text-sm font-semibold text-gray-900">Label</th>
-              <th scope="col" class="px-3 py-3.5 text-left text-sm font-semibold text-gray-900">Vm</th>
+              <th scope="col" class="px-3 py-3.5 text-left text-sm font-semibold text-gray-900">State</th>
```
Comment (fdr) [inline views/project/github.erb:35]: is this an intentional renaming? It seems orthogonal from patch purpose ("don't crash")

**Review by fdr — APPROVED** [review]
maybe there's a bit of debris here you could separate or something, but no fundamental issues.

File: views/project/github.erb (hunk 3 above)
Comment (enescakir) [reply views/project/github.erb:35]: This is another outcome of moving it to model as `display_state`

From the commit message:
> With recent changes, this state isn't only the state of virtual machine. So I moved to the runner model as "display_state" similar to the other models
